### ISSUE TITLE
Generated observations are not conformant to us-core-smokingstatus 4.0.0

### ISSUE THREAD

**Issue by lmsurpre** [synthetichealth/synthea#1034]
### What happened?

With `exporter.fhir.use_us_core_ig = true` (the default) Synthea generates smoking status observation resources with a profile of `http://hl7.org/fhir/us/core/StructureDefinition/us-core-smokingstatus` and a category like the following:
```
      "category": [ {
        "coding": [ {
          "system": "http://terminology.hl7.org/CodeSystem/observation-category",
          "code": "survey",
          "display": "survey"
        } ]
      } ]
```

This was valid in US Core 3.1.1 but in US Core 4.0.0 this profile now requires a fixed category code with value `social-history`.

https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-smokingstatus.html

### Relevant log output

```shell
{
    "fullUrl": "urn:uuid:9f1bbe32-5546-9a1b-6d83-7b6665b9639c",
    "resource": {
      "resourceType": "Observation",
      "id": "9f1bbe32-5546-9a1b-6d83-7b6665b9639c",
      "meta": {
        "profile": [ "http://hl7.org/fhir/us/core/StructureDefinition/us-core-smokingstatus" ]
      },
      "status": "final",
      "category": [ {
        "coding": [ {
          "system": "http://terminology.hl7.org/CodeSystem/observation-category",
          "code": "survey",
          "display": "survey"
        } ]
      } ],
      "code": {
        "coding": [ {
          "system": "http://loinc.org",
          "code": "72166-2",
          "display": "Tobacco smoking status NHIS"
        } ],
        "text": "Tobacco smoking status NHIS"
      },
      "subject": {
        "reference": "urn:uuid:ae055079-e543-04cf-5966-f16f1f203fe2"
      },
      "encounter": {
        "reference": "urn:uuid:85c40277-c7d0-e522-3bf6-92ab84e36219"
      },
      "effectiveDateTime": "2014-08-09T03:26:12-04:00",
      "issued": "2014-08-09T03:26:12.026-04:00",
      "valueCodeableConcept": {
        "coding": [ {
          "system": "http://snomed.info/sct",
          "code": "266919005",
          "display": "Never smoker"
        } ],
        "text": "Never smoker"
      }
    },
    "request": {
      "method": "POST",
      "url": "Observation"
    }
  }
```

**Comment by jawalonoski** [comment]
This requires migrating our unit tests from using US Core 3.1.1 to US Core 4.0.0.

This is probably long overdue... :grimacing:

**Comment by jawalonoski** [comment]
Fixed with #1036
